src/expr.rs

```diff
@@ -74,7 +74,13 @@ fn format_expr(expr: &ast::Expr,
         }
         ast::ExprKind::Call(ref callee, ref args) => {
             let inner_span = mk_sp(callee.span.hi, expr.span.hi);
-            rewrite_call(context, &**callee, args, inner_span, shape)
+            if context.inside_macro {
+                let mut context = context.clone();
+                context.inside_macro = false;
+                rewrite_call(&context, &**callee, args, inner_span, shape)
```

Comment:
I think we should be doing this at a higher level in the stack. What I mean is that we set this true so as not to insert/remove commas, so once we are walking deeper in the tree where changing commas, we should unset it. At the moment, we only check this in function calls, but one could imagine we check it for a different AST node and then we have a bug if we forget to unset the flag there too.